graphdatascience/tests/integration/test_coverage.py

```diff
@@ -63,9 +79,17 @@ def test_coverage(gds: GraphDataScience) -> None:
 
         # Check that each step of the string building is a valid object
         base = gds
-        for attr in endpoint_components:
+        for idx, attr in enumerate(endpoint_components):
             try:
                 base = getattr(base, attr)
                 assert base
+
+                if idx < len(endpoint_components) - 1:
+                    continue
+
+                assert callable(base)
+
+                if isinstance(base, UncallableNamespace):
+                    raise AssertionError()
             except Exception:
                 raise AssertionError(f"Could not find a client endpoint for the {server_endpoint} server endpoint")
```

Comment:
Some classes, the one's that should not be called, inherit from `UncallableNamespace`. They are "callable" in the Python sense because of how we do error handling in the library, but should not be called (hence the name of the class). Therefore we also have this if check. We can discuss it further on discord or whatever if you like

Reply:
The `idx` is used to only check the last part of the chain of attributes. For example: `gds.pageRank.stream` has three indices, and we only want to check that `gds.pageRank.stream` is callable, not `gds` or `gds.pageRank`. Hope that makes sense

Reply:
ah okay that makes sense. Maybe adding a comment on this line would help future readers as well :)

Reply:
I will add some comments

Reply:
done